### ISSUE TITLE
ENAMETOOLONG error thrown when converting big files

### ISSUE THREAD

**Issue by Graloth** [manuth/MarkdownConverter#1]
Decided to give the extension a try, but it seems to only want to generate HTML files.

This is the error I get when I try to convert it to PDF or PNG:
```
Error: spawnSync C:\Users\*private*\.vscode\extensions\manuth.markdown-converter-0.0.6\node_modules\phantomjs-prebuilt\lib\phantom\bin\phantomjs.exe ENAMETOOLONG
```
I have tried with file names as short as 3 letters, standard, no special characters, no spaces, etc. nothing works.

Like I mentioned, it works 100% with conversion to HTML.

No idea why this is happening, so I'm making an issue with it here, in case it's something that needs attention.

**Comment by manuth** [comment]
Hi there
I just gave it another try with the currently published version (0.0.6) and sadly I couldn't reproduce the error.
Can you please answer me some questions in order to help me finding the source of the problem?

The error you described actually only occurs when the file (phantomjs.exe in this case) is located in a too deep nested folder.

  - Is the \*private\*-part a deep nested folder-structure?
  - Is your destination-path (either the one you setted in the config (`markdownConverter.OutDir` or the folder which contains the markdown-file) an extremely deep nested one?
  - Are you using the most recent version? (0.0.2 or higher)

Thanks for your feedback

~manuth

**Comment by wBureshGitHub** [comment]
Hi, Manuel

Very good start!

However l have so (Linux, Debian, Stretch):

Error: spawnSync /home/w/.vscode/extensions/manuth.markdown-converter-0.0.6/node_modules/phantomjs-prebuilt/lib/phantom\bin\phantomjs.exe ENOENT
Open FileClose
InfoSuccessfully wrote the HTML-file to "/home/w/wWorks/buresh_ru/zDev-ib-ru/infoPravo/Law_3-includs/README.html".

Regards,
Vladimir

**Comment by manuth** [comment]
Hi Vladimir
Looks just like I messed the path separator up somehow.
I'll fix this asap and run my tests also on an Ubuntu-machine in order to ensure its working.
Can you please open up a new issue since Graloth's issue doesn't seem to be related to yours?

Thanks
~Manu

**Comment by manuth** [comment]
Hi, Vladimir - I opened up a new issue which describes your problem in order to keep this separate.
[MarkdownConverter Issue #2](https://github.com/manuth/MarkdownConverter/issues/2)

**Comment by manuth** [comment]
Please reopen the issue if you have futher troubles.
I'll close the issue. I guess you could solve it yourself since you didn't answer anymore.

**Comment by Graloth** [comment]
Sorry for the very late reply, hadn't had the time to check out your answer.  
I am currently using version 0.0.7, and still get the error.

Here are the answers for your questions:
  * The \*private\*-part is just my account name, so just a single folder.
  * The destination is in `C:\Projects\Personal\markdown-debugging`, so shouldn't be too deeply nested.
  * Yes, version 0.0.7 as of writing this.

The install path of phantomjs.exe does seem very deeply nested, since it mentions the path in the error, but I have no control over that, as it is where the extension is installed.

Any chance you can add a config option for a custom path to the phantomjs library? It should solve the issue if I just install phantomjs to the local project and then point the config option to that location, as the path then shouldn't be that deeply nested.

**Comment by manuth** [comment]
Hello Graloth

I could reproduce the error. Looks like it occurs when the file has a certain size.
I'll try to fix this somehow - thanks for notifing me about that!

~Manu

**Comment by manuth** [comment]
I finally could find the source of the problem!
It's because of Windows' argument size limitations.
I'm passing a JSON-string that represents the whole document (and its settings) as a console argument which causes the error.

I could fix the issue, so this improvement will be available in version 0.0.8
